mylar/webserve.py

```diff
@@ -4924,6 +4924,83 @@ def importReadlist(self, filename):
         raise cherrypy.HTTPRedirect("detailStoryArc?StoryArcID=%s&StoryArcName=%s" % (storyarcid, storyarc))
     importReadlist.exposed = True
 
+    def GenerateCBLReadingList(self, StoryArcID=None):
+
+        myDB = db.DBConnection()
+
+        if StoryArcID:
+            ReadingArc = myDB.select("SELECT * FROM storyarcs WHERE StoryArcID=? AND (Manual IS NOT 'deleted' OR Manual IS NULL) ORDER BY ReadingOrder", [StoryArcID])
+
+            if ReadingArc is None:
+                logger.info("Couldn't find Story Arc to use for reading list!")
+            else:
+                arcName = ReadingArc[0]['StoryArc']
+                logger.fdebug('Generating CBL file for ' + arcName)
+
+                # !Replace root_path variable
+                readingListDest = os.path.join(mylar.CONFIG.DESTINATION_DIR, 'ReadingLists')
+                filename = arcName + ".cbl"
+                fullPath = os.path.join(readingListDest, filename)
+
+                if not os.path.exists(readingListDest):
+                    logger.fdebug('making directory: %s' % readingListDest)
+                    filechecker.validateAndCreateDirectory(readingListDest, True)
+
+                logger.fdebug('Opening file: ' + fullPath)
+
+                with open(fullPath, 'w') as file:
+
+                    line1 = "<?xml version=\"1.0\"?>"
+                    line2 = "<ReadingList xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\">"
+                    line3 = "<Name>" + escape(arcName) + "</Name>"
```

Comment:
escape in this reference will cause a traceback when generating the cbl. ``escape`` isn't loaded anywhere and so it cannot be referenced. There is no need to escape, with the encoding set to utf-8. Just remove it and do not use the + to join the   values - do it like ``"<Name>%s</Name>" % (arcName)``